Redo Dockerfile generation

kayabaNerve opened the pull request:
PR opened as this should be equivalent to develop.

Further work is justified before merging.

kayabaNerve committed:
Delete all Dockerfile fragments, add new orchestator to generate Dockerfiles

Enables greater templating.

Also delete the unmaintained kubernetes folder *for now*. This should be
restored in the future.

kayabaNerve committed:
Use Dockerfiles from the orchestator

kayabaNerve committed:
Ignore Dockerfiles in the git repo

kayabaNerve committed:
Remove CI job to check Dockerfiles are as expected now that they're no longer committed

kayabaNerve committed:
Remove old Dockerfiles from repo

kayabaNerve committed:
Use Debian for monero-wallet-rpc

kayabaNerve committed:
Remove replace_cmds for proper usage of entry-dev

Consolidates ports a bit.

Updates serai-docker-tests from "compose" to "build".

kayabaNerve committed:
Only write a new dockerfile if it's distinct

Preserves the updated time metadata.

kayabaNerve committed:
Update serai-docker-tests

kayabaNerve committed:
Correct the path Dockerfiles are built from

kayabaNerve committed:
Correct inclusion of orchestration folder in Docker builds

kayabaNerve committed:
Correct debug/release flagging in the cargo command

Apparently, --debug isn't an effective NOP yet an error.

kayabaNerve committed:
Correct path used to run the Serai node within a Dockerfile

kayabaNerve committed:
Correct path in Monero Dockerfile

kayabaNerve committed:
Attempt storing monerod in /usr/bin

kayabaNerve committed:
Use sudo to move into /usr/bin in CI

kayabaNerve committed:
Correct 18.3.0 to 18.3.1

kayabaNerve committed:
Escape * with quotes

kayabaNerve committed:
Update deny.toml, ADD orchestration in runtime Dockerfile

kayabaNerve committed:
Add --detach to the Monero GH CI

kayabaNerve committed:
Diversify dockerfiles by network

kayabaNerve committed:
Fixes to network-diversified orchestration

kayabaNerve committed:
Bitcoin and Monero testnet scripts

kayabaNerve committed:
Permissions and tweaks

kayabaNerve committed:
Flatten scripts folders

kayabaNerve committed:
Add missing folder specification to Monero Dockerfile

kayabaNerve committed:
Have monero-wallet-rpc specify the monerod login

kayabaNerve committed:
Have the Docker CMD specify env variables inserted at time of Dockerfile generation

They're overrideable with the global enviornment as for tests. This enables
variable generation in orchestrator and output to productionized Docker files
without creating a life-long file within the Docker container.

kayabaNerve committed:
Don't add Dockerfiles into Docker containers now that they have secrets

Solely add the source code for them as needed to satisfy the workspace bounds.

kayabaNerve committed:
Download arm64 Monero on arm64

kayabaNerve committed:
Ensure constant host architecture when reproducibly building the wasm

Host architecture, for some reason, can effect the generated code despite the
target architecture always being foreign to the host architecture.

kayabaNerve committed:
Randomly generate infrastructure keys

kayabaNerve committed:
Have orchestrator generate a key, be able to create/start containers

kayabaNerve committed:
Ensure bash is used over sh

kayabaNerve committed:
Clean dated docs

kayabaNerve committed:
Change how quoting occurs

rlking commented on orchestration/dev/coins/monero/run.sh:
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
```
the alpine image doesn't have a bash by default

kayabaNerve replied:
Fun. Guess I'll consolidate to /sh :sweat_smile:  Thanks.

I pushed these changes as I went to bed, will review the accumulated CI errors now.

kayabaNerve committed:
Standardize to sh

kayabaNerve committed:
Have Docker test build the dev Dockerfiles

kayabaNerve committed:
Only key_gen once

kayabaNerve committed:
cargo update

Adds a patch for zstd and reconciles the breaking nightly change which just
occurred.

kayabaNerve committed:
Use a dedicated network for Serai

Also fixes SERAI_HOSTNAME passed to coordinator.

kayabaNerve committed:
Support providing a key over the env for the Serai node

kayabaNerve committed:
Enable and document running daemons for tests via serai-orchestrator

Has running containers under the dev network port forward the RPC ports.

kayabaNerve committed:
Use volumes for bitcoin/monero

kayabaNerve committed:
Use bitcoin's run.sh in GH CI

kayabaNerve committed:
Only use the volume for testnet (not dev)
